// @giphy/react-native-sdk/lib/commonjs/native/types.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ResizeMode = exports.GiphyThemePreset = exports.GiphyStickersColumnCount = exports.GiphyRendition = exports.GiphyRating = exports.GiphyMediaType = exports.GiphyFileExtension = exports.GiphyDirection = exports.GiphyContentType = exports.GiphyContentRequestType = exports.GiphyClipsRendition = void 0;
let GiphyThemePreset;
exports.GiphyThemePreset = GiphyThemePreset;

(function (GiphyThemePreset) {
  GiphyThemePreset["Automatic"] = "automatic";
  GiphyThemePreset["Dark"] = "dark";
  GiphyThemePreset["Light"] = "light";
})(GiphyThemePreset || (exports.GiphyThemePreset = GiphyThemePreset = {}));

let GiphyContentType;
exports.GiphyContentType = GiphyContentType;

(function (GiphyContentType) {
  GiphyContentType["Emoji"] = "emoji";
  GiphyContentType["Gif"] = "gif";
  GiphyContentType["Recents"] = "recents";
  GiphyContentType["Sticker"] = "sticker";
  GiphyContentType["Text"] = "text";
  GiphyContentType["Clips"] = "clips";
})(GiphyContentType || (exports.GiphyContentType = GiphyContentType = {}));

let GiphyMediaType;
exports.GiphyMediaType = GiphyMediaType;

(function (GiphyMediaType) {
  GiphyMediaType["Gif"] = "gif";
  GiphyMediaType["Sticker"] = "sticker";
  GiphyMediaType["Text"] = "text";
  GiphyMediaType["Video"] = "video";
})(GiphyMediaType || (exports.GiphyMediaType = GiphyMediaType = {}));

let GiphyRating;
exports.GiphyRating = GiphyRating;

(function (GiphyRating) {
  GiphyRating["G"] = "g";
  GiphyRating["PG"] = "pg";
  GiphyRating["PG13"] = "pg-13";
  GiphyRating["R"] = "r";
  GiphyRating["Unrated"] = "unrated";
  GiphyRating["Y"] = "y";
})(GiphyRating || (exports.GiphyRating = GiphyRating = {}));

let GiphyRendition;
exports.GiphyRendition = GiphyRendition;

(function (GiphyRendition) {
  GiphyRendition["Original"] = "original";
  GiphyRendition["OriginalStill"] = "original_still";
  GiphyRendition["Preview"] = "preview";
  GiphyRendition["Looping"] = "looping";
  GiphyRendition["FixedHeight"] = "fixed_height";
  GiphyRendition["FixedHeightStill"] = "fixed_height_still";
  GiphyRendition["FixedHeightDownsampled"] = "fixed_height_downsampled";
  GiphyRendition["FixedHeightSmall"] = "fixed_height_small";
  GiphyRendition["FixedHeightSmallStill"] = "fixed_height_small_still";
  GiphyRendition["FixedWidth"] = "fixed_width";
  GiphyRendition["FixedWidthStill"] = "fixed_width_still";
  GiphyRendition["FixedWidthDownsampled"] = "fixed_width_downsampled";
  GiphyRendition["FixedWidthSmall"] = "fixed_width_small";
  GiphyRendition["FixedWidthSmallStill"] = "fixed_width_small_still";
  GiphyRendition["Downsized"] = "downsized";
  GiphyRendition["DownsizedSmall"] = "downsized_small";
  GiphyRendition["DownsizedMedium"] = "downsized_medium";
  GiphyRendition["DownsizedLarge"] = "downsized_large";
  GiphyRendition["DownsizedStill"] = "downsized_still";
})(GiphyRendition || (exports.GiphyRendition = GiphyRendition = {}));

const GiphyClipsRendition = {
  Original: GiphyRendition.Original,
  OriginalStill: GiphyRendition.OriginalStill,
  FixedHeight: GiphyRendition.FixedHeight,
  FixedHeightStill: GiphyRendition.FixedHeightStill,
  FixedHeightDownsampled: GiphyRendition.FixedHeightDownsampled,
  FixedWidth: GiphyRendition.FixedWidth,
  FixedWidthStill: GiphyRendition.FixedWidthStill,
  FixedWidthDownsampled: GiphyRendition.FixedWidthDownsampled,
  DownsizedMedium: GiphyRendition.DownsizedMedium,
  DownsizedLarge: GiphyRendition.DownsizedLarge
};
exports.GiphyClipsRendition = GiphyClipsRendition;
let GiphyStickersColumnCount;
exports.GiphyStickersColumnCount = GiphyStickersColumnCount;

(function (GiphyStickersColumnCount) {
  GiphyStickersColumnCount[GiphyStickersColumnCount["Two"] = 2] = "Two";
  GiphyStickersColumnCount[GiphyStickersColumnCount["Three"] = 3] = "Three";
  GiphyStickersColumnCount[GiphyStickersColumnCount["Four"] = 4] = "Four";
})(GiphyStickersColumnCount || (exports.GiphyStickersColumnCount = GiphyStickersColumnCount = {}));

let GiphyFileExtension;
exports.GiphyFileExtension = GiphyFileExtension;

(function (GiphyFileExtension) {
  GiphyFileExtension["GIF"] = "gif";
  GiphyFileExtension["MP4"] = "mp4";
  GiphyFileExtension["WebP"] = "webp";
})(GiphyFileExtension || (exports.GiphyFileExtension = GiphyFileExtension = {}));

let GiphyDirection;
exports.GiphyDirection = GiphyDirection;

(function (GiphyDirection) {
  GiphyDirection["Horizontal"] = "horizontal";
  GiphyDirection["Vertical"] = "vertical";
})(GiphyDirection || (exports.GiphyDirection = GiphyDirection = {}));

let GiphyContentRequestType;
exports.GiphyContentRequestType = GiphyContentRequestType;

(function (GiphyContentRequestType) {
  GiphyContentRequestType["Trending"] = "trending";
  GiphyContentRequestType["Search"] = "search";
  GiphyContentRequestType["Emoji"] = "emoji";
  GiphyContentRequestType["Recents"] = "recents";
  GiphyContentRequestType["Animate"] = "animate";
})(GiphyContentRequestType || (exports.GiphyContentRequestType = GiphyContentRequestType = {}));

let ResizeMode;
exports.ResizeMode = ResizeMode;

(function (ResizeMode) {
  ResizeMode["Center"] = "center";
  ResizeMode["Contain"] = "contain";
  ResizeMode["Cover"] = "cover";
  ResizeMode["Stretch"] = "stretch";
})(ResizeMode || (exports.ResizeMode = ResizeMode = {}));
