// @giphy/react-native-sdk/lib/module/native/types.js
export let GiphyThemePreset;

(function (GiphyThemePreset) {
  GiphyThemePreset["Automatic"] = "automatic";
  GiphyThemePreset["Dark"] = "dark";
  GiphyThemePreset["Light"] = "light";
})(GiphyThemePreset || (GiphyThemePreset = {}));

export let GiphyContentType;

(function (GiphyContentType) {
  GiphyContentType["Emoji"] = "emoji";
  GiphyContentType["Gif"] = "gif";
  GiphyContentType["Recents"] = "recents";
  GiphyContentType["Sticker"] = "sticker";
  GiphyContentType["Text"] = "text";
  GiphyContentType["Clips"] = "clips";
})(GiphyContentType || (GiphyContentType = {}));

export let GiphyMediaType;

(function (GiphyMediaType) {
  GiphyMediaType["Gif"] = "gif";
  GiphyMediaType["Sticker"] = "sticker";
  GiphyMediaType["Text"] = "text";
  GiphyMediaType["Video"] = "video";
})(GiphyMediaType || (GiphyMediaType = {}));

export let GiphyRating;

(function (GiphyRating) {
  GiphyRating["G"] = "g";
  GiphyRating["PG"] = "pg";
  GiphyRating["PG13"] = "pg-13";
  GiphyRating["R"] = "r";
  GiphyRating["Unrated"] = "unrated";
  GiphyRating["Y"] = "y";
})(GiphyRating || (GiphyRating = {}));

export let GiphyRendition;

(function (GiphyRendition) {
  GiphyRendition["Original"] = "original";
  GiphyRendition["OriginalStill"] = "original_still";
  GiphyRendition["Preview"] = "preview";
  GiphyRendition["Looping"] = "looping";
  GiphyRendition["FixedHeight"] = "fixed_height";
  GiphyRendition["FixedHeightStill"] = "fixed_height_still";
  GiphyRendition["FixedHeightDownsampled"] = "fixed_height_downsampled";
  GiphyRendition["FixedHeightSmall"] = "fixed_height_small";
  GiphyRendition["FixedHeightSmallStill"] = "fixed_height_small_still";
  GiphyRendition["FixedWidth"] = "fixed_width";
  GiphyRendition["FixedWidthStill"] = "fixed_width_still";
  GiphyRendition["FixedWidthDownsampled"] = "fixed_width_downsampled";
  GiphyRendition["FixedWidthSmall"] = "fixed_width_small";
  GiphyRendition["FixedWidthSmallStill"] = "fixed_width_small_still";
  GiphyRendition["Downsized"] = "downsized";
  GiphyRendition["DownsizedSmall"] = "downsized_small";
  GiphyRendition["DownsizedMedium"] = "downsized_medium";
  GiphyRendition["DownsizedLarge"] = "downsized_large";
  GiphyRendition["DownsizedStill"] = "downsized_still";
})(GiphyRendition || (GiphyRendition = {}));

export const GiphyClipsRendition = {
  Original: GiphyRendition.Original,
  OriginalStill: GiphyRendition.OriginalStill,
  FixedHeight: GiphyRendition.FixedHeight,
  FixedHeightStill: GiphyRendition.FixedHeightStill,
  FixedHeightDownsampled: GiphyRendition.FixedHeightDownsampled,
  FixedWidth: GiphyRendition.FixedWidth,
  FixedWidthStill: GiphyRendition.FixedWidthStill,
  FixedWidthDownsampled: GiphyRendition.FixedWidthDownsampled,
  DownsizedMedium: GiphyRendition.DownsizedMedium,
  DownsizedLarge: GiphyRendition.DownsizedLarge
};
export let GiphyStickersColumnCount;

(function (GiphyStickersColumnCount) {
  GiphyStickersColumnCount[GiphyStickersColumnCount["Two"] = 2] = "Two";
  GiphyStickersColumnCount[GiphyStickersColumnCount["Three"] = 3] = "Three";
  GiphyStickersColumnCount[GiphyStickersColumnCount["Four"] = 4] = "Four";
})(GiphyStickersColumnCount || (GiphyStickersColumnCount = {}));

export let GiphyFileExtension;

(function (GiphyFileExtension) {
  GiphyFileExtension["GIF"] = "gif";
  GiphyFileExtension["MP4"] = "mp4";
  GiphyFileExtension["WebP"] = "webp";
})(GiphyFileExtension || (GiphyFileExtension = {}));

export let GiphyDirection;

(function (GiphyDirection) {
  GiphyDirection["Horizontal"] = "horizontal";
  GiphyDirection["Vertical"] = "vertical";
})(GiphyDirection || (GiphyDirection = {}));

export let GiphyContentRequestType;

(function (GiphyContentRequestType) {
  GiphyContentRequestType["Trending"] = "trending";
  GiphyContentRequestType["Search"] = "search";
  GiphyContentRequestType["Emoji"] = "emoji";
  GiphyContentRequestType["Recents"] = "recents";
  GiphyContentRequestType["Animate"] = "animate";
})(GiphyContentRequestType || (GiphyContentRequestType = {}));

export let ResizeMode;

(function (ResizeMode) {
  ResizeMode["Center"] = "center";
  ResizeMode["Contain"] = "contain";
  ResizeMode["Cover"] = "cover";
  ResizeMode["Stretch"] = "stretch";
})(ResizeMode || (ResizeMode = {}));
